### ISSUE TITLE
Whirlpool "died suddenly"

### ISSUE THREAD

**Issue by shyfire131** [sparrowwallet/sparrow#930]
Long time user here. I'm experiencing a very strange issue trying to use Whirlpool. I've spent a lot of time investigating this issue, so sorry about the lengthy writeup.

**Setup**
- Sparrow 1.7.4 (issue was also present on 1.7.3).
- macOS Ventura 13.3
- Single Native Segwit P2WPKH software wallet open (I also use a ColdCard wallet, but it's currently not an open tab in Sparrow)
- Mixing over VPN 
- [Sparrow settings] Connected to private Electrum server (via ssh tunnel on LAN i.e. Sparrow has 127.0.0.1 as the IP of the electrs server)
- [Sparrow settings] Use SSL disabled 
- [Sparrow settings] Use Proxy disabled (VPN is sufficient for me)
- As I typed this, I tried all of the above over Tor and the symptoms are identical

**Symptoms**
- No remixes for 4 days, despite mix control button switching to "Stop Mixing" state
- UTXO registration just cycles through UTXOs ( 
- Check logs, see single line:
`2023-04-05 18:40:02,940 ERROR [Thread-49] c.s.s.i.c.CardTransport [null:-1] Card initialization error, response was 0x6a82`
- Enable debug logging, see 300 lines, but it's all just `EatWhatYouKill` type stuff, happy to share those logs 

**Debugging Actions**
- Click "Stop Mixing" in Sparrow
- Button turns to "Start Mixing" after a while, mixing is deactivated
- Errors still appearing in logfile every 30 seconds:

```
2023-04-05 18:57:33,586 ERROR [Thread-65] c.s.s.i.c.CardTransport [null:-1] Card initialization error, response was 0x6a82
2023-04-05 18:58:03,551 ERROR [Thread-69] c.s.s.i.c.CardTransport [null:-1] Card initialization error, response was 0x6a82
2023-04-05 18:58:33,538 ERROR [Thread-71] c.s.s.i.c.CardTransport [null:-1] Card initialization error, response was 0x6a82
```

- Quit Sparrow
- Relaunch Sparrow
- Don't start mixing
- Error still appearing in logfile every 30 seconds
- Changed from VPN setup to Tor
- Error still appearing in logfile every 30 seconds
- Closed all wallet files
- Error STOPS
- Open a single wallet file (again, P2WPKH software wallet)
- Errors start appearing again every 30 seconds

----

It appears as if this c.s.s.i.c.CardTransport is a driver of sorts for the ColdCard, the error is certainly originating on [Line 50](https://github.com/sparrowwallet/sparrow/blob/master/src/main/java/com/sparrowwallet/sparrow/io/ckcard/CardTransport.java#L50) of `CardTransport.java` but my CC Sparrow wallet file is closed... so it's almost like Sparrow is still trying to open a channel to the coldcard despite the wallet being closed?

Happy to help debugging, just let me know

**Comment by craigraw** [comment]
Thanks for the detailed report.

The `CardTransport` error is actually related to a card reader linked to your machine (for Tapsigner/Satscard support), and unrelated to Whirlpool or the Coldcard. It's obviously not ideal that it is producing so much logging - I'll take a look at reducing that. Specifically, Sparrow is detecting some kind of card reader, but failing to initialize it, and logging errors. This shouldn't affect anything, so it can be safely ignored. If you can unplug the card reader, that should stop the errors.

It's certainly possible to get no mixes for a few days - sometimes randomization does strange things. Sparrow cycling through UTXOs (the Whirlpool client does so every hour) is also entirely normal and healthy. Depending on the pool, I've also experienced slower remixing this year, probably related to the fee pressure putting people off making transactions (in the hope that the mempool size reduces). Anecdotally, the smaller pools are also getting more mixes that the larger ones of late. 

In summary, I don't think anything is wrong - but please comment if you see anything else you feel is unusual. Also - have you considered using Sparrow Terminal - it can make mixing lower time preference if you don't need to spend soon.
